src/sentry/eventstream/kafka/protocol.py

```diff
@@ -33,23 +33,19 @@ def get_task_kwargs_for_insert(operation, event_data, task_state=None):
         # Rust (re)normalization here again would be too slow.
         event_data["data"] = EventDict(event_data["data"], skip_renormalization=True)
 
-        kwargs = {
-            "event": Event(
-                **{
-                    name: event_data[name]
-                    for name in [
-                        "group_id",
-                        "event_id",
-                        "project_id",
-                        "message",
-                        "platform",
-                        "datetime",
-                        "data",
-                    ]
-                }
-            ),
-            "primary_hash": event_data["primary_hash"],
-        }
+        event = SnubaEvent(
+            {
+                "group_id": event_data["group_id"],
+                "event_id": event_data["event_id"],
+                "project_id": event_data["project_id"],
+                "message": event_data["message"],
+                "platform": event_data["platform"],
+                "timestamp": event_data["datetime"].strftime("%Y-%m-%dT%H:%M:%S"),
```

Comment:
This works because snubaevent.datetime resolves to timestamp. Is that correct ?

Reply:
Yeah that's right. The test that changed shows the microseconds get dropped now. I think that's ok though since it's in line with what we return for datetime elsewhere in the API